router/batchrouter/asyncdestinationmanager/lytics_bulk_upload/lytics_test.go

```diff
@@ -0,0 +1,162 @@
+package lyticsBulkUpload_test
+
+import (
+	"context"
+	"net/http"
+	"reflect"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"go.uber.org/mock/gomock"
+
+	backendconfig "github.com/rudderlabs/rudder-server/backend-config"
+	mocks "github.com/rudderlabs/rudder-server/mocks/router/lytics_bulk_upload"
+	"github.com/rudderlabs/rudder-server/router/batchrouter/asyncdestinationmanager/common"
+	lyticsBulkUpload "github.com/rudderlabs/rudder-server/router/batchrouter/asyncdestinationmanager/lytics_bulk_upload"
+)
+
+var destination = &backendconfig.DestinationT{
+	ID:   "1",
+	Name: "LYTICS_BULK_UPLOAD",
+	DestinationDefinition: backendconfig.DestinationDefinitionT{
+		Name: "LYTICS_BULK_UPLOAD",
+	},
+	Config: map[string]interface{}{
+		"lyticsAccountId":  "1234",
+		"lyticsApiKey":     "1234567",
+		"lyticsStreamName": "test",
+		"timestampField":   "timestamp",
+		"streamTraitsMapping": []map[string]string{
+			{
+				"rudderProperty": "Email",
+				"lyticsProperty": "Email",
+			},
+		},
+	},
+	Enabled:     true,
+	WorkspaceID: "1",
+}
+
+func TestNewManagerSuccess(t *testing.T) {
+	manager, err := lyticsBulkUpload.NewManager(destination)
+	assert.NoError(t, err)
+	assert.NotNil(t, manager)
+	assert.Equal(t, "LYTICS_BULK_UPLOAD", destination.Name)
+}
+
+func TestUpload(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockUploader := mocks.NewMockUploader(ctrl)
+
+	expectedOutput := common.AsyncUploadOutput{
+		ImportingJobIDs: []int64{1, 2, 3},
+	}
+
+	mockUploader.EXPECT().Upload(gomock.Any()).Return(expectedOutput).Times(1)
```

Comment:
we can't mock this as this is being tested

Reply:
updated